### ISSUE TITLE
Search indexing traverses the wrong panel include files when building a site outside its directory

### ISSUE THREAD

**Issue by yamgent** [MarkBind/markbind#381]
**MarkBind version:** 1.9.2

Running `markbind serve test/test_site` on the root of the repository causes the following errors:

```
warning: ENOENT: no such file or directory, open 'C:\Users\Tan Wang Leng\Documents\GitHub\yamgent\markbind\test\test_site\sub_site\_markbind\variables.
md'
warning: Error: ENOENT: no such file or directory, open 'C:\Users\Tan Wang Leng\Documents\GitHub\yamgent\markbind\test\testPanels\PanelNormalSource._in
clude_.html'
warning: Error: ENOENT: no such file or directory, open 'C:\Users\Tan Wang Leng\Documents\GitHub\yamgent\markbind\test\testPanels\PanelNormalSource._in
clude_.html'
error: ENOENT: no such file or directory, open 'C:\Users\Tan Wang Leng\Documents\GitHub\yamgent\markbind\test\testPanels\PanelNormalSource._include_.ht
ml'
```

These errors do not surface when Travis execute tests, because they serve directly inside the `test/test_site` folder.

**Comment by acjh** [comment]
`markbind serve test/test_site` was fixed by #202

**Comment by yamgent** [comment]
> `markbind serve test/test_site` was fixed by #202

Thanks for mentioning that. We now separately traverse through the files when doing search indexing for panels (since dynamic panels' contents won't be inside the HTML file when the site is built). Judging from the error messages above, that might be the source of the problem.

**Comment by acjh** [comment]
I see. Can you improve the title of this issue?

**Comment by Chng-Zhi-Xuan** [comment]
Looks like this has been fixed in the current version of MarkBind (v2.4.2). Although I am not sure which PR is responsible.

![s1](https://user-images.githubusercontent.com/31084833/58154459-39efac00-7ca4-11e9-8b2c-a8204cbb5a16.PNG)

The remaining warning is from this snippet from `Site.js#collectUserDefinedVariablesMap`, where there is no `variables.md` in `test_site/sub_site/`
``` Javascript
    try {
      const userDefinedVariablesPath = path.resolve(base, USER_VARIABLES_PATH);
      content = fs.readFileSync(userDefinedVariablesPath, 'utf8');
    } catch (e) {
      content = '';
      logger.warn(e.message);
    }
```

**Comment by openorclose** [comment]
I tried it and got the same result too.

> Looks like this has been fixed in the current version of MarkBind (v2.4.2). Although I am not sure which PR is responsible.
> 
> ![s1](https://user-images.githubusercontent.com/31084833/58154459-39efac00-7ca4-11e9-8b2c-a8204cbb5a16.PNG)
> 
> The remaining warning is from this snippet in `Site.js#collectUserDefinedVariablesMap`, where there is no `variables.md` in `test_site/sub_site/`
> 
> ```js
>     try {
>       const userDefinedVariablesPath = path.resolve(base, USER_VARIABLES_PATH);
>       content = fs.readFileSync(userDefinedVariablesPath, 'utf8');
>     } catch (e) {
>       content = '';
>       logger.warn(e.message);
>     }
> ```

**Comment by Chng-Zhi-Xuan** [comment]
I'll close this issue for now, if it resurfaces we can reopen it.
